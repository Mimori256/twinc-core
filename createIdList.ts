export const createIdList = (fileContent: string, isFromKdBAlt: boolean) => {
  const normalize = (s: string) => s.replace(/"/g, "").trim();

  const idList = isFromKdBAlt
    ? fileContent
      .split("\n")
      .map((x) => x.split(",")[0])
      .map(normalize)
      .filter((x) => x !== "" && x !== "科目番号")
      .filter((x, i, self) => self.indexOf(x) === i)
    : fileContent
      .split("\n")
      .map(normalize)
      .filter((x) => x !== "" && x !== "科目番号")
      .filter((x, i, self) => self.indexOf(x) === i);

  return idList;
};
