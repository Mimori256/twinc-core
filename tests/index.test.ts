import { readFileSync } from "node:fs";
import { createIdList } from "../createIdList";
import kdb from "../data/sample-2023.json";
import parseCSV from "../parse";

describe("createIdList from TWINS data", () => {
  const fileContent = readFileSync("tests/twins-data.csv", "utf-8");
  const idList = createIdList(fileContent, false);
  it("should return an array", () => {
    expect(Array.isArray(idList)).toBe(true);
  });
  it("should return an valid array", () => {
    expect(idList.every((x) => typeof x === "string")).toBe(true);
    expect(idList.every((x) => x.length === 7)).toBe(true);
  });
  it("should match the expected result", () => {
    expect(idList).toMatchSnapshot();
  });
});

describe("createIdList from KdB alt data", () => {
  const fileContent = readFileSync("tests/kdb-alt.csv", "utf-8");
  const idList = createIdList(fileContent, true);
  it("should return an array", () => {
    expect(Array.isArray(idList)).toBe(true);
  });
  it("should return an valid array", () => {
    expect(idList.every((x) => typeof x === "string")).toBe(true);
    expect(idList.every((x) => x.length === 7)).toBe(true);
  });
  it("should match the expected result", () => {
    expect(idList).toMatchSnapshot();
  });
});

describe("parseCSV", () => {
  const idList = [
    "GC55301",
    "GC51401",
    "GC50501",
    "GC41203",
    "GC53401",
    "GC55201",
    "",
  ];
  const ics = `${parseCSV(idList, kdb, true, false)}END:VCALENDAR`;
  it("should return a string", () => {
    expect(typeof ics).toBe("string");
  });
  it("should match the expected result", () => {
    expect(ics).toMatchSnapshot();
  });
});
