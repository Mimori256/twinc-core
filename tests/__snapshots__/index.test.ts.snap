// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createIdList from KdB alt data should match the expected result 1`] = `
Array [
  "GC22201",
  "GC41103",
  "GC41203",
  "GC51701",
  "GC51901",
  "GC52701",
  "GC53401",
  "GC55401",
]
`;

exports[`createIdList from TWINS data should match the expected result 1`] = `
Array [
  "GC51701",
  "GC51901",
  "GC52701",
  "GC53401",
  "GC55401",
]
`;

exports[`parseCSV should match the expected result 1`] = `
"BEGIN:VCALENDAR
PRODID:-//gam0022//TwinC 1.0//EN
VERSION:2.0
CALSCALE:GREGORIAN
METHOD:PUBLISH
X-WR-CALNAME:授業時間割
X-WR-TIMEZONE:Asia/Tokyo
X-WR-CALDESC:授業時間割
BEGIN:VTIMEZONE
TZID:Asia/Tokyo
X-LIC-LOCATION:Asia/Tokyo
BEGIN:STANDARD
TZOFFSETFROM:+0900
TZOFFSETTO:+0900
TZNAME:JST
DTSTART:19700102T000000
END:STANDARD
END:VTIMEZONE
BEGIN:VEVENT
DTSTART;TZID=Asia/Tokyo:20251007T084000
DTEND;TZID=Asia/Tokyo:20251007T095500
RRULE:FREQ=WEEKLY;UNTIL=20251219T130000Z;BYDAY=TU
EXDATE:20251013T084000,20251016T084000,20251103T084000,20251104T084000,20251105T084000,20251107T084000,20251124T084000,20251125T084000,20251126T084000,20251127T084000,20251128T084000,
DTSTAMP:20250408T000000
CREATED:20250408T000000
DESCRIPTION:GE62401, GE72701と同一。 英語で授業。 G科目. 対面(オンライン併用型) 情報メディア創成学類生はGC55301を、知識情報システム主専攻生はGE72701を、それ以外の学生はGE62401を履修すること。
LAST-MODIFIED:20250408T000000
LOCATION:7A103
SEQUENCE:0
STATUS:CONFIRMED
SUMMARY:Machine Learning and Information Retrieval
TRANSP:OPAQUE
END:VEVENT
BEGIN:VEVENT
DTSTART;TZID=Asia/Tokyo:20251105T084000
DTEND;TZID=Asia/Tokyo:20251105T095500
DTSTAMP:20250408T000000
CREATED:20250408T000000
DESCRIPTION:GE62401, GE72701と同一。 英語で授業。 G科目. 対面(オンライン併用型) 情報メディア創成学類生はGC55301を、知識情報システム主専攻生はGE72701を、それ以外の学生はGE62401を履修すること。
LAST-MODIFIED:20250408T000000
LOCATION:7A103
SEQUENCE:0
STATUS:CONFIRMED
SUMMARY:Machine Learning and Information Retrieval
TRANSP:OPAQUE
END:VEVENT
BEGIN:VEVENT
DTSTART;TZID=Asia/Tokyo:20251007T101000
DTEND;TZID=Asia/Tokyo:20251007T112500
RRULE:FREQ=WEEKLY;UNTIL=20251219T130000Z;BYDAY=TU
EXDATE:20251013T101000,20251016T101000,20251103T101000,20251104T101000,20251105T101000,20251107T101000,20251124T101000,20251125T101000,20251126T101000,20251127T101000,20251128T101000,
DTSTAMP:20250408T000000
CREATED:20250408T000000
DESCRIPTION:GE62401, GE72701と同一。 英語で授業。 G科目. 対面(オンライン併用型) 情報メディア創成学類生はGC55301を、知識情報システム主専攻生はGE72701を、それ以外の学生はGE62401を履修すること。
LAST-MODIFIED:20250408T000000
LOCATION:7A103
SEQUENCE:0
STATUS:CONFIRMED
SUMMARY:Machine Learning and Information Retrieval
TRANSP:OPAQUE
END:VEVENT
BEGIN:VEVENT
DTSTART;TZID=Asia/Tokyo:20251105T101000
DTEND;TZID=Asia/Tokyo:20251105T112500
DTSTAMP:20250408T000000
CREATED:20250408T000000
DESCRIPTION:GE62401, GE72701と同一。 英語で授業。 G科目. 対面(オンライン併用型) 情報メディア創成学類生はGC55301を、知識情報システム主専攻生はGE72701を、それ以外の学生はGE62401を履修すること。
LAST-MODIFIED:20250408T000000
LOCATION:7A103
SEQUENCE:0
STATUS:CONFIRMED
SUMMARY:Machine Learning and Information Retrieval
TRANSP:OPAQUE
END:VEVENT
BEGIN:VEVENT
DTSTART;TZID=Asia/Tokyo:20251007T151500
DTEND;TZID=Asia/Tokyo:20251007T163000
RRULE:FREQ=WEEKLY;UNTIL=20251219T130000Z;BYDAY=TU
EXDATE:20251013T151500,20251016T151500,20251103T151500,20251104T151500,20251105T151500,20251107T151500,20251124T151500,20251125T151500,20251126T151500,20251127T151500,20251128T151500,
DTSTAMP:20250408T000000
CREATED:20250408T000000
DESCRIPTION:対面
LAST-MODIFIED:20250408T000000
LOCATION:7A105
SEQUENCE:0
STATUS:CONFIRMED
SUMMARY:システム運用・管理
TRANSP:OPAQUE
END:VEVENT
BEGIN:VEVENT
DTSTART;TZID=Asia/Tokyo:20251105T151500
DTEND;TZID=Asia/Tokyo:20251105T163000
DTSTAMP:20250408T000000
CREATED:20250408T000000
DESCRIPTION:対面
LAST-MODIFIED:20250408T000000
LOCATION:7A105
SEQUENCE:0
STATUS:CONFIRMED
SUMMARY:システム運用・管理
TRANSP:OPAQUE
END:VEVENT
BEGIN:VEVENT
DTSTART;TZID=Asia/Tokyo:20251001T084000
DTEND;TZID=Asia/Tokyo:20251001T095500
RRULE:FREQ=WEEKLY;UNTIL=20251219T130000Z;BYDAY=WE
EXDATE:20251013T084000,20251016T084000,20251103T084000,20251104T084000,20251105T084000,20251107T084000,20251124T084000,20251125T084000,20251126T084000,20251127T084000,20251128T084000,
DTSTAMP:20250408T000000
CREATED:20250408T000000
DESCRIPTION:「情報通信概論」(2018年度以前入学者対象)、もしくは「コンピュータネットワーク」(2019年度以降入学者対象)を履修していることが望ましい。 対面
LAST-MODIFIED:20250408T000000
LOCATION:ユニオン講義室
SEQUENCE:0
STATUS:CONFIRMED
SUMMARY:通信ネットワーク
TRANSP:OPAQUE
END:VEVENT
BEGIN:VEVENT
DTSTART;TZID=Asia/Tokyo:20251001T101000
DTEND;TZID=Asia/Tokyo:20251001T112500
RRULE:FREQ=WEEKLY;UNTIL=20251219T130000Z;BYDAY=WE
EXDATE:20251013T101000,20251016T101000,20251103T101000,20251104T101000,20251105T101000,20251107T101000,20251124T101000,20251125T101000,20251126T101000,20251127T101000,20251128T101000,
DTSTAMP:20250408T000000
CREATED:20250408T000000
DESCRIPTION:「情報通信概論」(2018年度以前入学者対象)、もしくは「コンピュータネットワーク」(2019年度以降入学者対象)を履修していることが望ましい。 対面
LAST-MODIFIED:20250408T000000
LOCATION:ユニオン講義室
SEQUENCE:0
STATUS:CONFIRMED
SUMMARY:通信ネットワーク
TRANSP:OPAQUE
END:VEVENT
BEGIN:VEVENT
DTSTART;TZID=Asia/Tokyo:20251001T121500
DTEND;TZID=Asia/Tokyo:20251001T133000
RRULE:FREQ=WEEKLY;UNTIL=20260205T130000Z;BYDAY=WE
EXDATE:20241014T121500,20241016T121500,20241104T121500,20241105T121500,20241107T121500,20241108T121500,20241125T121500,20241126T121500,20241127T121500,20241128T121500,20241129T121500,20241226T121500,20241227T121500,20241230T121500,20241231T121500,20250101T121500,20250102T121500,20250103T121500,20250113T121500,20250116T121500,20250117T121500,20250120T121500,20250122T121500,20250204T121500,20250207T121500,20250211T121500,20250212T121500,20250213T121500,20250214T121500,20250217T121500,
DTSTAMP:20250408T000000
CREATED:20250408T000000
DESCRIPTION:同一テーマが複数学期で開講されていても、2度以上選択することはできない。 対面(オンライン併用型)
LAST-MODIFIED:20250408T000000
LOCATION:7B106,7C202,クリエイティブメディアラボ
SEQUENCE:0
STATUS:CONFIRMED
SUMMARY:情報メディア実験B
TRANSP:OPAQUE
END:VEVENT
BEGIN:VEVENT
DTSTART;TZID=Asia/Tokyo:20251001T134500
DTEND;TZID=Asia/Tokyo:20251001T150000
RRULE:FREQ=WEEKLY;UNTIL=20260205T130000Z;BYDAY=WE
EXDATE:20241014T134500,20241016T134500,20241104T134500,20241105T134500,20241107T134500,20241108T134500,20241125T134500,20241126T134500,20241127T134500,20241128T134500,20241129T134500,20241226T134500,20241227T134500,20241230T134500,20241231T134500,20250101T134500,20250102T134500,20250103T134500,20250113T134500,20250116T134500,20250117T134500,20250120T134500,20250122T134500,20250204T134500,20250207T134500,20250211T134500,20250212T134500,20250213T134500,20250214T134500,20250217T134500,
DTSTAMP:20250408T000000
CREATED:20250408T000000
DESCRIPTION:同一テーマが複数学期で開講されていても、2度以上選択することはできない。 対面(オンライン併用型)
LAST-MODIFIED:20250408T000000
LOCATION:7B106,7C202,クリエイティブメディアラボ
SEQUENCE:0
STATUS:CONFIRMED
SUMMARY:情報メディア実験B
TRANSP:OPAQUE
END:VEVENT
BEGIN:VEVENT
DTSTART;TZID=Asia/Tokyo:20251003T151500
DTEND;TZID=Asia/Tokyo:20251003T163000
RRULE:FREQ=WEEKLY;UNTIL=20260205T130000Z;BYDAY=FR
EXDATE:20241014T151500,20241016T151500,20241104T151500,20241105T151500,20241107T151500,20241108T151500,20241125T151500,20241126T151500,20241127T151500,20241128T151500,20241129T151500,20241226T151500,20241227T151500,20241230T151500,20241231T151500,20250101T151500,20250102T151500,20250103T151500,20250113T151500,20250116T151500,20250117T151500,20250120T151500,20250122T151500,20250204T151500,20250207T151500,20250211T151500,20250212T151500,20250213T151500,20250214T151500,20250217T151500,
DTSTAMP:20250408T000000
CREATED:20250408T000000
DESCRIPTION:同一テーマが複数学期で開講されていても、2度以上選択することはできない。 対面(オンライン併用型)
LAST-MODIFIED:20250408T000000
LOCATION:7B106,7C202,クリエイティブメディアラボ
SEQUENCE:0
STATUS:CONFIRMED
SUMMARY:情報メディア実験B
TRANSP:OPAQUE
END:VEVENT
BEGIN:VEVENT
DTSTART;TZID=Asia/Tokyo:20260115T151500
DTEND;TZID=Asia/Tokyo:20260115T163000
DTSTAMP:20250408T000000
CREATED:20250408T000000
DESCRIPTION:同一テーマが複数学期で開講されていても、2度以上選択することはできない。 対面(オンライン併用型)
LAST-MODIFIED:20250408T000000
LOCATION:7B106,7C202,クリエイティブメディアラボ
SEQUENCE:0
STATUS:CONFIRMED
SUMMARY:情報メディア実験B
TRANSP:OPAQUE
END:VEVENT
BEGIN:VEVENT
DTSTART;TZID=Asia/Tokyo:20251003T164500
DTEND;TZID=Asia/Tokyo:20251003T180000
RRULE:FREQ=WEEKLY;UNTIL=20260205T130000Z;BYDAY=FR
EXDATE:20241014T164500,20241016T164500,20241104T164500,20241105T164500,20241107T164500,20241108T164500,20241125T164500,20241126T164500,20241127T164500,20241128T164500,20241129T164500,20241226T164500,20241227T164500,20241230T164500,20241231T164500,20250101T164500,20250102T164500,20250103T164500,20250113T164500,20250116T164500,20250117T164500,20250120T164500,20250122T164500,20250204T164500,20250207T164500,20250211T164500,20250212T164500,20250213T164500,20250214T164500,20250217T164500,
DTSTAMP:20250408T000000
CREATED:20250408T000000
DESCRIPTION:同一テーマが複数学期で開講されていても、2度以上選択することはできない。 対面(オンライン併用型)
LAST-MODIFIED:20250408T000000
LOCATION:7B106,7C202,クリエイティブメディアラボ
SEQUENCE:0
STATUS:CONFIRMED
SUMMARY:情報メディア実験B
TRANSP:OPAQUE
END:VEVENT
BEGIN:VEVENT
DTSTART;TZID=Asia/Tokyo:20260115T164500
DTEND;TZID=Asia/Tokyo:20260115T180000
DTSTAMP:20250408T000000
CREATED:20250408T000000
DESCRIPTION:同一テーマが複数学期で開講されていても、2度以上選択することはできない。 対面(オンライン併用型)
LAST-MODIFIED:20250408T000000
LOCATION:7B106,7C202,クリエイティブメディアラボ
SEQUENCE:0
STATUS:CONFIRMED
SUMMARY:情報メディア実験B
TRANSP:OPAQUE
END:VEVENT
BEGIN:VEVENT
DTSTART;TZID=Asia/Tokyo:20251002T121500
DTEND;TZID=Asia/Tokyo:20251002T133000
RRULE:FREQ=WEEKLY;UNTIL=20251219T130000Z;BYDAY=TH
EXDATE:20251013T121500,20251016T121500,20251103T121500,20251104T121500,20251105T121500,20251107T121500,20251124T121500,20251125T121500,20251126T121500,20251127T121500,20251128T121500,
DTSTAMP:20250408T000000
CREATED:20250408T000000
DESCRIPTION:GE70301と同一。 実務経験教員. 対面 情報メディア創成学類生はGC53401を,それ以外の学生はGE70301を履修すること
LAST-MODIFIED:20250408T000000
LOCATION:ユニオン講義室
SEQUENCE:0
STATUS:CONFIRMED
SUMMARY:ディジタルドキュメント
TRANSP:OPAQUE
END:VEVENT
BEGIN:VEVENT
DTSTART;TZID=Asia/Tokyo:20251002T134500
DTEND;TZID=Asia/Tokyo:20251002T150000
RRULE:FREQ=WEEKLY;UNTIL=20251219T130000Z;BYDAY=TH
EXDATE:20251013T134500,20251016T134500,20251103T134500,20251104T134500,20251105T134500,20251107T134500,20251124T134500,20251125T134500,20251126T134500,20251127T134500,20251128T134500,
DTSTAMP:20250408T000000
CREATED:20250408T000000
DESCRIPTION:GE70301と同一。 実務経験教員. 対面 情報メディア創成学類生はGC53401を,それ以外の学生はGE70301を履修すること
LAST-MODIFIED:20250408T000000
LOCATION:ユニオン講義室
SEQUENCE:0
STATUS:CONFIRMED
SUMMARY:ディジタルドキュメント
TRANSP:OPAQUE
END:VEVENT
BEGIN:VEVENT
DTSTART;TZID=Asia/Tokyo:20251003T121500
DTEND;TZID=Asia/Tokyo:20251003T133000
RRULE:FREQ=WEEKLY;UNTIL=20251219T130000Z;BYDAY=FR
EXDATE:20251013T121500,20251016T121500,20251103T121500,20251104T121500,20251105T121500,20251107T121500,20251124T121500,20251125T121500,20251126T121500,20251127T121500,20251128T121500,
DTSTAMP:20250408T000000
CREATED:20250408T000000
DESCRIPTION:GE62501, GE73101と同一。 英語で授業。 G科目. 対面 情報メディア創成学類生はGC55201を、知識情報システム主専攻生はGE73101を、それ以外の学生はGE62501を履修すること。 GE60701「レファレンスサイエンス」を修得済みの者は履修不可
LAST-MODIFIED:20250408T000000
LOCATION:7A210
SEQUENCE:0
STATUS:CONFIRMED
SUMMARY:Human Information Interaction
TRANSP:OPAQUE
END:VEVENT
BEGIN:VEVENT
DTSTART;TZID=Asia/Tokyo:20251003T134500
DTEND;TZID=Asia/Tokyo:20251003T150000
RRULE:FREQ=WEEKLY;UNTIL=20251219T130000Z;BYDAY=FR
EXDATE:20251013T134500,20251016T134500,20251103T134500,20251104T134500,20251105T134500,20251107T134500,20251124T134500,20251125T134500,20251126T134500,20251127T134500,20251128T134500,
DTSTAMP:20250408T000000
CREATED:20250408T000000
DESCRIPTION:GE62501, GE73101と同一。 英語で授業。 G科目. 対面 情報メディア創成学類生はGC55201を、知識情報システム主専攻生はGE73101を、それ以外の学生はGE62501を履修すること。 GE60701「レファレンスサイエンス」を修得済みの者は履修不可
LAST-MODIFIED:20250408T000000
LOCATION:7A210
SEQUENCE:0
STATUS:CONFIRMED
SUMMARY:Human Information Interaction
TRANSP:OPAQUE
END:VEVENT
BEGIN:VEVENT
DTSTART;VALUE=DATE:20250410
DTEND;VALUE=DATE:20250411
DTSTAMP:20220408T000000
CREATED:20220408T000000
STATUS:CONFIRMED
TRANSP:TRANSPARENT
SUMMARY:春A事前登録締め切り日
END:VEVENT
BEGIN:VEVENT
DTSTART;VALUE=DATE:20250425
DTEND;VALUE=DATE:20250426
DTSTAMP:20220408T000000
CREATED:20220408T000000
STATUS:CONFIRMED
TRANSP:TRANSPARENT
SUMMARY:春A履修登録締切日
END:VEVENT
BEGIN:VEVENT
DTSTART;VALUE=DATE:20250513
DTEND;VALUE=DATE:20250514
DTSTAMP:20220408T000000
CREATED:20220408T000000
STATUS:CONFIRMED
TRANSP:TRANSPARENT
SUMMARY:春B事前登録締め切り日
END:VEVENT
BEGIN:VEVENT
DTSTART;VALUE=DATE:20250522
DTEND;VALUE=DATE:20250523
DTSTAMP:20220408T000000
CREATED:20220408T000000
STATUS:CONFIRMED
TRANSP:TRANSPARENT
SUMMARY:春A期末試験
END:VEVENT
BEGIN:VEVENT
DTSTART;VALUE=DATE:20250529
DTEND;VALUE=DATE:20250530
DTSTAMP:20220408T000000
CREATED:20220408T000000
STATUS:CONFIRMED
TRANSP:TRANSPARENT
SUMMARY:春B履修登録締切日
END:VEVENT
BEGIN:VEVENT
DTSTART;VALUE=DATE:20250624
DTEND;VALUE=DATE:20250625
DTSTAMP:20220408T000000
CREATED:20220408T000000
STATUS:CONFIRMED
TRANSP:TRANSPARENT
SUMMARY:春C事前登録締め切り日
END:VEVENT
BEGIN:VEVENT
DTSTART;VALUE=DATE:20250710
DTEND;VALUE=DATE:20250711
DTSTAMP:20220408T000000
CREATED:20220408T000000
STATUS:CONFIRMED
TRANSP:TRANSPARENT
SUMMARY:春C履修登録締切日
END:VEVENT
BEGIN:VEVENT
DTSTART;VALUE=DATE:20250922
DTEND;VALUE=DATE:20250923
DTSTAMP:20220408T000000
CREATED:20220408T000000
STATUS:CONFIRMED
TRANSP:TRANSPARENT
SUMMARY:秋A事前登録締め切り日
END:VEVENT
BEGIN:VEVENT
DTSTART;VALUE=DATE:20251014
DTEND;VALUE=DATE:20251015
DTSTAMP:20220408T000000
CREATED:20220408T000000
STATUS:CONFIRMED
TRANSP:TRANSPARENT
SUMMARY:秋A履修登録締切日
END:VEVENT
BEGIN:VEVENT
DTSTART;VALUE=DATE:20251029
DTEND;VALUE=DATE:20251030
DTSTAMP:20220408T000000
CREATED:20220408T000000
STATUS:CONFIRMED
TRANSP:TRANSPARENT
SUMMARY:秋B事前登録締め切り日
END:VEVENT
BEGIN:VEVENT
DTSTART;VALUE=DATE:20251010
DTEND;VALUE=DATE:20251011
DTSTAMP:20220408T000000
CREATED:20220408T000000
STATUS:CONFIRMED
TRANSP:TRANSPARENT
SUMMARY:秋A期末試験
END:VEVENT
BEGIN:VEVENT
DTSTART;VALUE=DATE:20251117
DTEND;VALUE=DATE:20251118
DTSTAMP:20220408T000000
CREATED:20220408T000000
STATUS:CONFIRMED
TRANSP:TRANSPARENT
SUMMARY:秋B履修登録締切日
END:VEVENT
BEGIN:VEVENT
DTSTART;VALUE=DATE:20251217
DTEND;VALUE=DATE:20251218
DTSTAMP:20220408T000000
CREATED:20220408T000000
STATUS:CONFIRMED
TRANSP:TRANSPARENT
SUMMARY:秋C事前登録締め切り日
END:VEVENT
BEGIN:VEVENT
DTSTART;VALUE=DATE:20260115
DTEND;VALUE=DATE:20260116
DTSTAMP:20220408T000000
CREATED:20220408T000000
STATUS:CONFIRMED
TRANSP:TRANSPARENT
SUMMARY:秋C履修登録締切日
END:VEVENT
END:VCALENDAR"
`;
